Darknet importer: unable to import clasification configuration file darknet19.cfg

Using one of the cfg file from darknet repo. (darknet-19 for classification not detection).

https://github.com/AlexeyAB/darknet/blob/master/cfg/darknet19.cfg

opencv/modules/dnn/src/darknet/darknet_io.cpp:678: error: (-212:Parsing error) Unknown layer type: sse in function 'ReadDarknetFromCfgStream'

Seems to be in cost layer...

[cost]
type=sse

I remove it but then when I try

`cv::dnn::Net net = cv::dnn::readNet(modelConfiguration, modelWeights);

    net.setPreferableBackend(GetAnalyticsParameters()->GetDNNBackend());
    net.setPreferableTarget(GetAnalyticsParameters()->GetDNNTarget());

    std::cout << net.dump() << std::endl;`

terminate called after throwing an instance of 'cv::Exception'
  what():  OpenCV(4.2.0-dev) /opencv/modules/dnn/src/dnn.cpp:3781: error: (-2:Unspecified error) Requested set input in function 'dump'

@alalek , @dkurt 
while parsing
```
[cost]
type=sse
```
the parameter `type` is interpreted as layer type. producing error. 
`darknet_io.cpp:762: error: (-212:Parsing error) Unknown layer type: sse in function 'ReadDarknetFromCfgStream'`
Because here we are storing layer type with key "type".
https://github.com/opencv/opencv/blob/102ef39a27bfc277d82045e5c5e0d4b1251294df/modules/dnn/src/darknet/darknet_io.cpp#L552-L559
And here "type" parameter is getting stored with key "type" itself. 
https://github.com/opencv/opencv/blob/102ef39a27bfc277d82045e5c5e0d4b1251294df/modules/dnn/src/darknet/darknet_io.cpp#L562-L575
So ignoring while parsing until "[" i.e. next layer is seen is better option ? Am I right ?

--- a/modules/dnn/src/darknet/darknet_io.cpp
+++ b/modules/dnn/src/darknet/darknet_io.cpp
@@ -556,7 +556,7 @@ namespace cv {
                             const size_t layer_type_size = line.find("]") - 1;
                             CV_Assert(layer_type_size < line.size());
                             std::string layer_type = line.substr(1, layer_type_size);
-                            net->layers_cfg[layers_counter]["type"] = layer_type;
+                            net->layers_cfg[layers_counter]["layer_type"] = layer_type;
                         }
                         break;
                     default:
@@ -599,7 +599,7 @@ namespace cv {
                 for (it_type i = net->layers_cfg.begin(); i != net->layers_cfg.end(); ++i) {
                     ++layers_counter;
                     std::map<std::string, std::string> &layer_params = i->second;
-                    std::string layer_type = layer_params["type"];
+                    std::string layer_type = layer_params["layer_type"];
 
                     if (layer_type == "convolutional")
                     {
@@ -682,7 +682,7 @@ namespace cv {
                         else
                             setParams.setConcat(layers_vec.size(), layers_vec.data());
                     }
-                    else if (layer_type == "dropout")
+                    else if (layer_type == "dropout" || layer_type == "cost")
                     {
                         setParams.setIdentity(layers_counter-1);
                     }
@@ -806,7 +806,7 @@ namespace cv {
                     ++darknet_layers_counter;
                     ++cv_layers_counter;
                     std::map<std::string, std::string> &layer_params = i->second;
-                    std::string layer_type = layer_params["type"];
+                    std::string layer_type = layer_params["layer_type"];
 
                     if (layer_type == "convolutional" || layer_type == "connected")
                     {

